pkg/daemon/daemon.go

```diff
@@ -1290,12 +1295,55 @@ func (dn *Daemon) completeUpdate(desiredConfigName string) error {
 		return err
 	}
 
+	if err := dn.removeUpdateInProgressTaint(); err != nil {
+		return err
+	}
+
 	dn.logSystem("Update completed for config %s and node has been successfully uncordoned", desiredConfigName)
 	dn.recorder.Eventf(getNodeRef(dn.node), corev1.EventTypeNormal, "Uncordon", fmt.Sprintf("Update completed for config %s and node has been uncordoned", desiredConfigName))
 
 	return nil
 }
 
+func (dn *Daemon) removeUpdateInProgressTaint() error {
+	return clientretry.RetryOnConflict(commonconstants.NodeUpdateBackoff, func() error {
+		oldNode, err := dn.kubeClient.CoreV1().Nodes().Get(context.TODO(), dn.name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		oldData, err := json.Marshal(oldNode)
+		if err != nil {
+			return err
+		}
+
+		newNode := oldNode.DeepCopy()
+		if newNode.Spec.Taints == nil {
+			newNode.Spec.Taints = []corev1.Taint{}
+		}
+
+		// Get the index of NodeUpdateInProgressTaint.
+		taintIndex := 0
+		for i, taint := range newNode.Spec.Taints {
+			if taint.MatchTaint(commonconstants.NodeUpdateInProgressTaint) {
+				taintIndex = i
+			}
+		}
+		// Remove the NodeUpdateInProgressTaint.
+		newNode.Spec.Taints = append(newNode.Spec.Taints[:taintIndex], newNode.Spec.Taints[taintIndex+1:]...)
```

Comment:
If `newNode.Spec.Taints = []corev1.Taint{}` still is the case here, wouldn't this panic?

Reply:
[this][1] would be one way to avoid that panic.

[1]: https://github.com/openshift/machine-config-operator/pull/2686#discussion_r764383642

Reply:
Ah right. Alternatively if the taint is not found, we could probably return early above.